gms/gossiper.cc

```diff
@@ -702,10 +702,21 @@ future<> gossiper::remove_endpoint(inet_address endpoint, permit_id pid) {
 
     auto state = get_endpoint_state_ptr(endpoint);
 
-    bool was_alive;
+    bool was_alive = false;
     co_await mutate_live_and_unreachable_endpoints([endpoint, &was_alive] (gossiper& g) {
-        was_alive = g._live_endpoints.erase(endpoint);
-        g._unreachable_endpoints.erase(endpoint);
+        return live_and_unreachable_endpoints{
```

Comment:
Maybe mutate_live_and_unreachable_endpoints() should perform the copy and call its lambda with the copy?

Losing optimization of folding the filtering into the copy is worth the extra clarity IMO.

Reply:
That will definitely simplify the flow, and in this case which is not on the hot path maybe the optimization is indeed not worth it.

Reply:
done in v2